.github/workflows/ci.yaml

```
@@ -315,7 +315,7 @@ jobs:
         with:
           python-version: ${{ env.PYTHON_VERSION }}
       - name: install ruff
-        run: python -m pip install ruff
+        run: python -m pip install ruff==0.0.291
```

@youknowone Perhaps inserting a comment here would be helpful in case someone wants to unpin this dependency.
```suggestion
        run: python -m pip install ruff==0.0.291 # astral-sh/ruff#7778
```